src/util.c

```diff
@@ -582,35 +582,36 @@ int dcc_remove_if_exists(const char *fname)
 
 int dcc_which(const char *command, char **out)
 {
-    char *loc = NULL, *_loc, *path, *t;
-    int ret;
+    char *loc = NULL, *path, *t;
+    int len, ret;
 
     path = getenv("PATH");
     if (!path)
         return -ENOENT;
+
     do {
-        if (strstr(path, "distcc"))
-            continue;
-        /* emulate strchrnul() */
         t = strchr(path, ':');
         if (!t)
-            t = path + strlen(path);
-        _loc = realloc(loc, t - path + 1 + strlen(command) + 1);
-        if (!_loc) {
-            free(loc);
-            return -ENOMEM;
+           len = strlen(path);
+        else
+            len = t - path;
+        loc = realloc(loc, len + strlen(command) + 2);
+        if (!loc) {
+             free(loc);
```

Comment:
It's not a huge deal because this will only happen on ENOMEM, which should be rare, but: the `free` seems pointless, since you know `loc` is null here. You could keep the old one in another variable, as the old code does.